googletest/src/gtest-death-test.cc

```diff
@@ -1357,6 +1357,9 @@ static pid_t ExecDeathTestSpawnChild(char* const* argv, int close_fd) {
     const auto stack_size = static_cast<size_t>(getpagesize());
     // MMAP_ANONYMOUS is not defined on Mac, so we use MAP_ANON instead.
     void* const stack = mmap(nullptr, stack_size, PROT_READ | PROT_WRITE,
+#    if GTEST_OS_LINUX
+                             MAP_GROWSDOWN | // Needed for stack auto-resizing
```

Comment:
This patch works for me on my x86_64 system.
Without patch, I can reproduce problem.

Above the changed code, there is:
```
    static const bool stack_grows_down = StackGrowsDown();
```
I suspect that passing `MAP_GROWSDOWN` unconditionally to `mmap()` might cause problems on relatively rare architectures (e.g. HPPA) where stack grows up.
It is probably safer to use:
```
+#    if GTEST_OS_LINUX
+                             (stack_grows_down ? MAP_GROWSDOWN : 0) | // Needed for stack auto-resizing
+#    endif
```

Reply:
Good call. Updated PR.